app/src/main/java/ca/wheresthebus/service/GtfsRealtimeHelper.kt

```diff
@@ -28,11 +25,19 @@ class GtfsRealtimeHelper {
         private val client = OkHttpClient()
         private const val GTFS_API_URL = "https://gtfsapi.translink.ca/v3/gtfsrealtime?apikey=${ca.wheresthebus.BuildConfig.GTFS_KEY}"
 
-        suspend fun getBusTimes(stopId: StopId, routeId: RouteId): List<Duration> {
+        suspend fun getBusTimes(stopsInfo: List<Pair<StopId, RouteId>>): List<List<Duration>> {
             return try {
                 val feedMessage = callGtfsRealtime()
-                val busTimes = grabBusTimes(feedMessage, stopId, routeId)
-                convertBusTimes(busTimes)
+                stopsInfo.map { (stopId, routeId) ->
+                    try {
+                        val busTimes = grabBusTimes(feedMessage, stopId, routeId)
```

Comment:
This could probably be optimized so that it takes in the list and returns all the appropriate bus times at once, but that needs heavy refactoring of the grabBusTimes function, but it's fast enough already